ext/RastersStatsBaseExt/sample.jl

```diff
@@ -0,0 +1,89 @@
+Rasters.sample(x::RA.RasterStackOrArray, n::Integer; kw...) = Rasters.sample(Random.GLOBAL_RNG, x, n; kw...)
+@inline function Rasters.sample(
+    rng::Random.AbstractRNG, x::RA.RasterStackOrArray, n::Integer; 
+    geometry=(X,Y),
+    index = false, 
+    names=RA._names(x), 
+    name=names, 
+    skipmissing=false,
+    replace=true, 
+    ordered=false, 
+    weights=nothing, 
+    weightstype::Type{<:StatsBase.AbstractWeights}=StatsBase.Weights
+)
+    na = DD._astuple(name)
+    geometry, geometrytype, dims = _geometrytype(x, geometry)
+ #   x = x isa RA.AbstractRasterStack ? x[na] : x
+    _sample(rng, x, n;
+        dims=DD.dims(x, RA.DEFAULT_POINT_ORDER),
+        names=NamedTuple{na}(na),
+        # These keywords are converted to _True/_False for type stability later on
+        # The @inline above helps constant propagation of the Bools
+        geometry=geometry,
+        geometrytype=geometrytype,  
+        index=_booltype(index), 
+        skipmissing=_booltype(skipmissing), 
+        weights,
+        replace, 
+        ordered,
+        weightstype
+    )
+
+end
+function _sample(
+    rng, x, n; 
+    dims, names::NamedTuple{K}, geometry, geometrytype, index, skipmissing, weights, replace, ordered, weightstype
+) where K
+    indices = sample_indices(rng, x, n, skipmissing, weights, replace, ordered, weightstype)
+    points = DimPoints(dims)
+    T = RA._rowtype(x, geometrytype; geometry, index, skipmissing, skipinvalid = _True(), names)
+    x2 = x isa AbstractRasterStack ? x[K] : RasterStack(NamedTuple{K}((x,)))
+    _getindices(T, x2, points, indices)
+end
+function _getindices(::Type{T}, x, points, indices) where T
+    rows = Vector{T}(undef, size(indices))
+    for (i, I) in enumerate(indices)
+        rows[i] = _getindex(T, x, points, I)
+    end
+    return rows
+end
+
+_getindex(::Type{T}, x::AbstractRasterStack{<:Any, NT}, points, idx) where {T, NT} = 
+    RA._maybe_add_fields(T, NT(x[RA.commondims(idx, x)]), points[RA.commondims(idx, points)], val(idx))
```

Comment:
Explicitly using the eltype of the stack like this was the only way I could find in which this would compile away. But it looks somehow ugly and I don't know if there's a better way?

Reply:
Like what? Is this comment on the wrong line?